src/ngx_http_lua_worker_thread.c

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (C) Yichun Zhang (agentzh)
+ * Copyright (C) Jinhua Luo (kingluo)
+ * I hereby assign copyright in this code to the lua-nginx-module project,
+ * to be licensed under the same terms as the rest of the code.
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_worker_thread.h"
+#include "ngx_http_lua_util.h"
+
+
+#if (NGX_THREADS)
+
+
+#include <ngx_thread.h>
+#include <ngx_thread_pool.h>
+
+
+typedef struct ngx_http_lua_task_ctx_s {
+    lua_State                        *vm;
+    ngx_thread_mutex_t                mutex;
+    struct ngx_http_lua_task_ctx_s   *next;
+} ngx_http_lua_task_ctx_t;
+
+
+typedef struct {
+    ngx_http_lua_task_ctx_t *ctx;
+    ngx_http_lua_co_ctx_t   *wait_co_ctx;
+    ngx_http_cleanup_pt      cleanup;
+    void                    *cleanup_data;
+    int                      n_args;
+    int                      rc;
+    int                      is_abort;
+} ngx_http_lua_worker_thread_ctx_t;
+
+
+static  ngx_http_lua_task_ctx_t   dummy_ctx;
+static  ngx_http_lua_task_ctx_t  *ctxpool = &dummy_ctx;
+static  ngx_uint_t                worker_thread_vm_count;
+
+static ngx_http_lua_task_ctx_t *
+ngx_http_lua_get_task_ctx(lua_State *L, ngx_http_request_t *r)
+{
+    ngx_http_lua_task_ctx_t *ctx = NULL;
+
+    size_t           path_len;
+    const char      *path;
+    size_t           cpath_len;
+    const char      *cpath;
+    lua_State       *vm;
+
+    ngx_http_lua_main_conf_t    *lmcf;
+
+    lmcf = ngx_http_get_module_main_conf(r, ngx_http_lua_module);
+
+    worker_thread_vm_count++;
+
+    if (worker_thread_vm_count > lmcf->worker_thread_vm_pool_size) {
+        return NULL;
+    }
```

Comment:
we can  move such kind of limitation code into the code block after line 69:
https://github.com/openresty/lua-nginx-module/pull/1861/files#diff-0185abcb5f34ae5c702511951d834eb70e215b179311e788b680034aec185a0cR69

And we won't need such `--` then:
https://github.com/openresty/lua-nginx-module/pull/1861/files#diff-0185abcb5f34ae5c702511951d834eb70e215b179311e788b680034aec185a0cR114